docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,715 @@
+1. Allowing multiple secrets means that if we support some additional details such as annotations as 
+   [requested by customers](https://github.com/Azure/azure-service-operator/issues/1398), the common use case may result in customers duplicating the same
+   annotations (although possibly there would be no difference if they didn't, provided they were writing to the same secret for all keys).
+
+Personal note: I _think_ I like this one more.
+
+Conclusion: TBD
+
+### Other kinds of secrets
+
+See [other kinds of secrets in Azure](#other-kinds-of-secrets-in-azure) for examples of each of these types of secrets.
+
+**Get once Azure generated secrets**: These are secrets created by Azure and _only returned once_, usually as the response to a POST.
+
+This pattern doesn't seem to be very common. The current plan is to not support resources with this pattern. If we did need to support
+resources with this pattern there would be no way to avoid violating the [work well with GitOps](#work-well-with-gitops) goal, as when
+deploying this resource there would be no way to adopt secrets associated with it (they're GET-once). That might be ok for particular
+kinds of resources provided we document it though. 
+
+If there are large user requests for this sort of secret management we can support it similar to the `ListKeys` cases except that 
+the resources won't be movable without also cloning the secret.
+
+**"Secrets" created by Azure, and returned in the `GET`**: This is things like `InstrumentationKey`, or server endpoint URLs.
+
+Since these are returned in a `GET` they are not secret and are already being shown on Status. We will also manually classify some of them to be 
+included in secrets we write. That will include things like `endpoint`. This means that `endpoint` for a SQL Server would show up in two places,
+in the status _and_ possibly in the secret written by the operator (assuming that the user has instructed us to write the endpoint someplace).
+
+The main reason for writing this into two places is:
+1. We already have it in the `Status` and it doesn't make sense to remove it as it's part of the resource payload from Azure.
+2. We want the ability to put it into a Secret so that if users want to inject it into their pods as an environment variable it's easy to do so.
+   There is no good way to inject environment variables from CRD Status's.
+
+**Short-lived "tokens"**: Like Storage SAS.
+
+We will not support these sorts of secrets.
+
+
+### A special note on KeyVault
+We have an [open issue](https://github.com/Azure/azure-service-operator/issues/1894) asking for support to manage KeyVault secrets via ASO.
+
+We have to be _very_ careful about support for KeyVault secrets (or certs, keys, etc), as the whole point of KeyVault is as a secure place to store your 
+secrets.
+If you're instead creating those secrets via the operator then you have by definition also located the secret in Kubernetes, which somewhat defeats the purpose 
```

Comment:
Good point. Do we then create a KeyVault operator to essentially do the role of the secret store CSI driver, or is there anything else we can add?

Reply:
I don't know what the right story is here honestly. It kinda feels like, if you want keyvault _for security reasons_, then you by definition do not want your secrets in Kubernetes and so no Kubernetes based solution for storing (or creating) the secrets can work for you.

If you want KeyVault because Azure "makes" you for some case or another, then you might not care and an ASO KeyVault Key resource that takes a k8s `Secret` as input might make sense for that case. We'd need to be sure that case was common enough that we care about it though, and I am not 100% sure it is.

Note: This doesn't stop us from putting the secrets we get from Azure into KeyVault (which this proposal suggests support for). That's fine because we can bypass Kubernetes secrets entirely and then the user could use the KeyVault CSI driver to inject those same secrets that ASO created in KeyVault (via `ListKeys` on storage for example) into their pod.